store/tikv/gcworker/gc_worker.go

```diff
@@ -307,12 +270,41 @@ func (w *GCWorker) prepare() (bool, uint64, error) {
 	return doGC, safePoint, errors.Trace(err)
 }
 
+func (w *GCWorker) initTableValues() error {
+
+	// This is a static set of data observed in TiDB on a default install *prior* to moving tikv values to sysVars.
+	// It is no longer needed, but some examples in the wild and in manual pages use "UPDATE".
+	// This requires a row to exist for the operation to succeed. Because the ON DUPLICATE KEY UPDATE
+	// action is UPDATE variable_name = variable_name, it is a noop if the row already exists.
+
+	stmt := `INSERT HIGH_PRIORITY INTO mysql.tidb VALUES
+	('tikv_gc_enable', 'true', 'Current GC enable status'),
+	('tikv_gc_run_interval', '10m0s', 'GC run interval, at least 10m, in Go format.'),
+	('tikv_gc_life_time', '10m0s', 'All versions within life time will not be collected by GC, at least 10m, in Go format.')
+	ON DUPLICATE KEY
+	UPDATE variable_name = variable_name`
+
+	se := createSession(w.store)
+	defer se.Close()
+	_, err := se.Execute(context.Background(), stmt)
+	return err
+
+}
+
 func (w *GCWorker) checkPrepare(ctx context.Context) (bool, uint64, error) {
-	enable, err := w.checkGCEnable()
+	se := createSession(w.store)
+	defer se.Close()
+	enable, err := se.GetSessionVars().GlobalVarsAccessor.GetGlobalSysVar(variable.TiKVGCEnable)
```

Comment:
I think you should still check the mysql.tidb table instead of checking sysvars. The logic here requires to be exexuted in a transaction began at line 257. I think most part of GCWorker don't need to be changed, so do the unit tests. The new configuration way should be tested in additional test cases, and the old test cases should be kept to ensure we can still configure GC by the mysql.tidb table.

Reply:
The advantage of reading though mysql.tidb is that it will normalize/validate the values and restore the results when incorrect. There is some cleanup as well, since gcworker implements its own validation.

But I think your suggestion is easier for reviewers, especially if I can split it into two PRs (GCWorker can come later). So I will try this approach.